src/v2/controllers/vdeployment_controller/exec/reconciler.rs

```diff
@@ -0,0 +1,505 @@
+                old_vrs_list@.map_values(|vrs: VReplicaSet| vrs@).drop_last()
+            } else {
+                old_vrs_list@.map_values(|vrs: VReplicaSet| vrs@)
+            };
+            assert(pre_new_vrs_list == vrs_list@.map_values(|vrs: VReplicaSet| vrs@).take(idx as int).filter(new_spec_filter));
+            assert(pre_old_vrs_list == vrs_list@.map_values(|vrs: VReplicaSet| vrs@).take(idx as int).filter(old_spec_filter));
+            push_filter_and_filter_push(vrs_list@.map_values(|vrs: VReplicaSet| vrs@).take(idx as int), new_spec_filter, vrs_list@.map_values(|vrs: VReplicaSet| vrs@)[idx as int]);
+            push_filter_and_filter_push(vrs_list@.map_values(|vrs: VReplicaSet| vrs@).take(idx as int), old_spec_filter, vrs_list@.map_values(|vrs: VReplicaSet| vrs@)[idx as int]);
+            assert(vrs_list@.map_values(|vrs: VReplicaSet| vrs@).take(idx as int).push(vrs_list@.map_values(|vrs: VReplicaSet| vrs@)[idx as int])
+                   == vrs_list@.map_values(|vrs: VReplicaSet| vrs@).take(idx + 1 as int));
+            assert(new_spec_filter(vrs@) ==> new_vrs_list@.map_values(|vrs: VReplicaSet| vrs@) == pre_new_vrs_list.push(vrs@));
+            assert(old_spec_filter(vrs@) ==> old_vrs_list@.map_values(|vrs: VReplicaSet| vrs@) == pre_old_vrs_list.push(vrs@));
+        }
+
+        idx += 1;
+    }
+    assert(vrs_list@.map_values(|vrs: VReplicaSet| vrs@) == vrs_list@.map_values(|vrs: VReplicaSet| vrs@).take(vrs_list.len() as int));
+    (new_vrs_list, old_vrs_list)
+}
+
+// TODO
+// proof lemma_filter_old_and_new_vrs_match_model();
+fn make_replica_set(vd: &VDeployment) -> (vrs: VReplicaSet)
+    requires vd@.well_formed(),
+    ensures vrs@ == model_reconciler::make_replica_set(vd@),
+{
+    let pod_template_hash = vd.metadata().resource_version().unwrap();
+    let mut vrs = VReplicaSet::default();
+    vrs.set_metadata({
+        let mut metadata = ObjectMeta::default();
+        // concatenation of (String, String) not yet supported in Verus
+        metadata.set_name(vd.metadata().name().unwrap().concat("-").concat(pod_template_hash.as_str()));
+        metadata.set_namespace(vd.metadata().namespace().unwrap());
+        if vd.metadata().labels().is_some() {
+            metadata.set_labels(vd.metadata().labels().unwrap().clone());
+        }
+        metadata.set_owner_references(make_owner_references(vd));
+        metadata
+    });
+    let mut spec = VReplicaSetSpec::default();
+    if vd.spec().replicas().is_some() {
+        spec.set_replicas(vd.spec().replicas().unwrap());
+    }
+    let mut labels = vd.spec().template().unwrap().metadata().unwrap().labels().unwrap().clone();
+    labels.insert("pod-template-hash".to_string(), pod_template_hash);
+    let mut label_selector = LabelSelector::default();
+    label_selector.set_match_labels(labels);
+    spec.set_selector(label_selector);
+    let template = template_with_hash(vd);
+    spec.set_template(template);
+    vrs.set_spec(spec);
+    proof {
```

Comment:
Is this `proof` block really needed anymore now we fixed the dash/underscore discrepancy?